### ISSUE TITLE
Properties not working

### ISSUE THREAD

**Issue by ghnp5** [sirbrillig/phpcs-variable-analysis#124]
Hello,

First time using this, and somehow the properties seem to be ignored.

I added the rule VariableAnalysis.CodeAnalysis.VariableAnalysis, and definitely, I'm getting a number of valid reports on a file, but unfortunately I can't get the properties to work.

This is what I have (very similar to how it is in the README):

```xml
	<rule ref="VariableAnalysis.CodeAnalysis.VariableAnalysis">
		<properties>
			<property name="ignoreUnusedRegexp" value="/^_/"/>
			<property name="validUndefinedVariableNames" value="/pagination/"/>
		</properties>
	</rule>
```

However, I'm getting this report:

```
----------------------------------------------------------------------
FOUND 0 ERRORS AND 4 WARNINGS AFFECTING 3 LINES
----------------------------------------------------------------------
  889 | WARNING | Variable $pagination is undefined.
      |         | (VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable)
  917 | WARNING | Variable $_pagination is undefined.
      |         | (VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable)
 1434 | WARNING | Variable $row is undefined.
      |         | (VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable)
 1434 | WARNING | Variable $row is undefined.
      |         | (VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable)
----------------------------------------------------------------------
```

The first two should be ignored.

If I'm doing something wrong, please let me know.

**Comment by sirbrillig** [comment]
> 			<property name="validUndefinedVariableNames" value="/pagination/"/>

I think this should be `<property name="validUndefinedVariableNames" value="pagination"/>`

🤔  As for the `ignoreUnusedRegexp`... let me do some research to see if I can reproduce that issue.

**Comment by sirbrillig** [comment]
I wasn't able to reproduce this, but here's my minimal test case. Maybe you can compare it to your project and see what might be different.

The `phpcs.xml` file:
```xml
<?xml version="1.0"?>
<ruleset name="this-standard">
	<rule ref="VariableAnalysis"/>
	<rule ref="VariableAnalysis.CodeAnalysis.VariableAnalysis">
		<properties>
			<property name="ignoreUnusedRegexp" value="/^_/"/>
		</properties>
	</rule>
</ruleset>
```

The php file:

```php
<?php

function whatever($foo, $_bar) {
	echo $foo;
}
```

How I ran the command:

```
phpcs -s --standard=./phpcs.xml ./testfile.php
```

**Comment by ghnp5** [comment]
Hello,

Thank you. Regarding "validUndefinedVariableNames", it was just me working while I was sleepy, and I didn't realize I kept the slashes! I was aware it wasn't a regexp, but a space-separated list. :)
That fixed the issue.

Regarding "ignoreUnusedRegexp", once more I apologize... I was focused on "undefined", rather than "unused", as the property says in its name.
I was testing for undefined variables, not unused. It does work too!

That's perfect -- this can be closed, as everything is working correctly :)

Regards.
